stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/usecase/GetChannelControllerImpl.kt

```diff
@@ -20,13 +19,8 @@ public interface GetChannelController {
     public operator fun invoke(cid: String): Call<ChannelController>
 }
 
-internal class GetChannelControllerImpl(private val domainImpl: ChatDomainImpl) : GetChannelController {
-    override operator fun invoke(cid: String): Call<ChannelController> {
-        validateCid(cid)
-
-        val channelControllerImpl = domainImpl.channel(cid)
-        return CoroutineCall(domainImpl.scope) {
-            Result(channelControllerImpl)
-        }
-    }
+internal class GetChannelControllerImpl(private val offlineGetChannelController: OfflineGetChannelController) :
+    GetChannelController {
+    override operator fun invoke(cid: String): Call<ChannelController> =
+        offlineGetChannelController.invoke(cid).map(::ChannelControllerImpl)
```

Comment:
We need copying even those use cases that return simple `Call` because they have public interfaces from the `livedata` package and they must be kept, while versions from the `offline` package shouldn't have any dependency from `livedata`

Reply:
Got it, thank you! So we need these for now, and then if we remove the `livedata.ChatDomain.useCases` from public API, we can kill all the use cases in there, and just merge the mapping code into `livedata.ChatDomain` instead. That'll clean up a bunch of small files.